src/PostgREST/Auth.hs

```diff
@@ -117,7 +115,5 @@ parseSecret str =
 jwkFromSecret :: ByteString -> JWK
 jwkFromSecret key =
   fromKeyMaterial km
-    & jwkUse ?~ Sig
-    & jwkAlg ?~ JWSAlg HS256
```

Comment:
Hmm.. those are default?

Reply:
I'm not entirely sure. The values were marked as uncovered and all tests still pass without those lines. If those settings are the default, then the values would have still be used.

There must be something else. My assumption is, that those settings don't have any effect at all and are just neither used nor needed.

Reply:
I'm a bit scared of the change because the algorithm could default to `None`:

https://hackage.haskell.org/package/jose-jwt-0.8.0/docs/Jose-Jwa.html#t:JwsAlg

And it could be that the tests are just accepting any JWT.

Reply:
Well, if the value is not used at all, then it doesn't matter what we set it to.

My understanding is that a JWT does have the algorithm it was encoded with included. This is the header of a JWT:
```
{"typ":"JWT",
 "alg":"HS256"}
```

My best bet is, that those settings are only used when actually signing a JWT with this JWK. But we only ever use this JWK for verifying the signature of an existing JWT.

I further assume that the only reason we turn the secret into a JWK here is to be able to use real JWKs instead of text secrets easily with the same code path for the verification.

So basically, this is not a real JWK, but just a container for our secret to have the right type.

Reply:
Of course, it could also be, that we're just missing a proper test here and that's why it's uncovered. But I think that's unlikely, given that we do use properly signed JWTs in our tests and there is no config option that affects how we use the text secret, right? I.e. which encoding algorithm etc. - that's all part of application that actually signs the JWT.

Reply:
So I had a look at the code again: We are using the JWK that is created here in `attemptJwtClaims`. Here, we're using `defaultJWTValidationSettings` of which `defaultValidationSettings` is the main part:

https://hackage.haskell.org/package/jose-0.8.4/docs/Crypto-JOSE-JWS.html#v:defaultValidationSettings

There it says:
> The default validation settings.
> - All algorithms except "none" are acceptable.
> - All signatures must be valid (and there must be at least one signature.)

So the algorithm is in fact taken from the JWT, but "none" is not accepted.

Reply:
I also tracked down the signature verifying function: https://hackage.haskell.org/package/jose-0.8.4/docs/src/Crypto.JOSE.JWS.html#verifySig

It only reads `jwkMaterial` from the key and not `jwkAlg` or `jwkUse`. **If** we really want to limit the allowed algorithms for verifying JWTs, we'd need to change this line:

https://github.com/PostgREST/postgrest/blob/ab6f90aa78b5ff5b4643f9b3ac97ae4ff6b0b589/src/PostgREST/Auth.hs#L65

and change the allowed algorithms in those settings.

Reply:
> All algorithms except "none" are acceptable.
> It only reads jwkMaterial from the key and not jwkAlg or jwkUse

Great! Thanks for digging into this @wolfgangwalther.

So then I'm no longer worried about this change. It's safe to remove these lines.

> If we really want to limit the allowed algorithms for verifying JWTs, we'd need to change this line:

It looks fine to leave as it is for now. Good to know we can limit the algorithms though.